src/components/Clasification.jsx

```diff
@@ -0,0 +1,106 @@
+import { useSelector, useDispatch } from 'react-redux';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { fetchPokemon, fetchType } from '../redux/pokemon/pokemonSlice';
+import { filterType } from './Pokemon';
+import pokemonImg from '../assets/pokemon.png';
+
+const Clasification = () => {
+  let typesFiltered;
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const [loadingType, setLoadingType] = useState(true);
+  const [loadingPokemon, setLoadingPokemon] = useState(true);
+  const [search, setSearch] = useState('');
+
+  const pokemon = useSelector(
+    (state) => state.pokemon.pokemon,
+  );
+
+  const types = useSelector(
+    (state) => state.pokemon.type.results,
+  );
+
+  useEffect(() => {
+    dispatch(fetchType('pokemon')).then(() => {
+      setLoadingType(false);
+    });
+  }, [dispatch]);
+
+  useEffect(() => {
+    dispatch(fetchPokemon('pokemon')).then(() => {
+      setLoadingPokemon(false);
+    });
+  }, [dispatch]);
```

Comment:
- [ ] As mentioned by the previous :point_up: reviewer Your application should load the data from the API saved in the redux store for the first time, but after loading the application once OR successfully make the first fetch request. Your application should work even with the person moving from the `Home` page to the `Detail` page or vice versa by using the stored data from the redux Store. But in your case, your application first loads the data by making the fetch request to the URL then after loading the data, if I click on the `particular` class of Pokemon and then move back to the home page using the `back` arrow button. It makes the request again to the server, which is not a good idea and makes extra requests even if the data was loaded successfully at the beginning. Your job is to set up your dispatch action at a specific place like App.js to make the request only once and then a person can see the pages with data without any problem. This also helps you to make your application efficient and more performant but still, you didn't implement this change kindly implement this change :slightly_smiling_face:

Reply:
@aamirkhan2478 I'm trying to do it this way, but If I change my code I have problems in the tests and I can't solved.